traffic_ops/traffic_ops_golang/server/update.go

```diff
@@ -136,29 +141,166 @@ func UpdateHandler(w http.ResponseWriter, r *http.Request) {
+}
 
-	if _, err := tx.Exec(qry, qryVals...); err != nil {
-		return errors.New("executing: " + err.Error())
+// UpdateHandler implements an http handler that updates a server's config update and reval times.
+func UpdateHandlerV4(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, []string{"id-or-name"}, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
 	}
-	return nil
+	defer inf.Close()
+
+	idOrName := inf.Params["id-or-name"]
+	serverID, err := strconv.ParseInt(idOrName, 10, 64)
+	if err != nil {
+		id, ok, err := dbhelpers.GetServerIDFromName(idOrName, inf.Tx.Tx)
+		if err != nil {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, fmt.Errorf("getting server id from name '%v': %w", idOrName, err))
+			return
+		} else if !ok {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusNotFound, errors.New("server name '"+idOrName+"' not found"), nil)
+			return
+		}
+		serverID = int64(id)
+	}
+
+	cdnName, err := dbhelpers.GetCDNNameFromServerID(inf.Tx.Tx, serverID)
+	if err != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, err)
+		return
+	}
+	userErr, sysErr, statusCode := dbhelpers.CheckIfCurrentUserHasCdnLock(inf.Tx.Tx, string(cdnName), inf.User.UserName)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, statusCode, userErr, sysErr)
+		return
+	}
+
+	// Verify query string parameters
+	config_update_time, hasConfig_update_time := inf.Params["config_update_time"]
+	config_apply_time, hasConfig_apply_time := inf.Params["config_apply_time"]
+	revalidate_update_time, hasRevalidate_update_time := inf.Params["revalidate_update_time"]
+	revalidate_apply_time, hasRevalidate_apply_time := inf.Params["revalidate_apply_time"]
+	if !hasConfig_update_time && !hasConfig_apply_time && !hasRevalidate_update_time && !hasRevalidate_apply_time {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("must pass at least one query parameter: 'config_update_time', 'config_apply_time', 'revalidate_update_time', 'revalidate_apply_time'"), nil)
+		return
+	}
+
+	// Validate parameters before attempting to apply them (don't want to partially apply various status before an error)
+	var configUpdateTime time.Time
+	if hasConfig_update_time {
+		configUpdateTime, err = time.Parse(time.RFC3339Nano, config_update_time)
+		if err != nil {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("query parameter 'config_update_time' must be valid RFC3339Nano format"), nil)
+			return
+		}
+	}
+
+	var configApplyTime time.Time
+	if hasConfig_apply_time {
+		configApplyTime, err = time.Parse(time.RFC3339Nano, config_apply_time)
+		if err != nil {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("query parameter 'config_apply_time' must be valid RFC3339Nano format"), nil)
+			return
+		}
+	}
+
+	var revalUpdateTime time.Time
+	if hasRevalidate_update_time {
```

Comment:
I think you're right here. We should only be setting the apply times in this endpoints and queueing updates / revals elsewhere. I think T3C intends to use it as you said.

Buuuuut, I think this endpoint was ~abused~ misused in the past. Since it used to operate on booleans only, it functioned as an endpoint to do both queueing and dequeueing, as reflected in the t3c integration tests. Should we refactor this to only allow _apply_ times? And thusly refactor the T3C logic (specifically the tests) to use a different approach to queueing updates/revals?